examples/action_dann_lightn/model.py

```diff
@@ -60,49 +64,202 @@ def get_feat_extractor(model_name, num_classes, num_channels):
+            else:
+                logging.info("No SELayer.")
+                feature_network = mc3(rgb=False, flow=True, pretrained=True)
+            class_feature_dim = 512
+            dmn_feature_dim = class_feature_dim
+
+        else:
+            raise ValueError("Unsupported model: {}".format(model_name))
+
+    elif image_modality == "joint":
+        if model_name == "I3D":
+            rgb_pretrained_model = "rgb_imagenet"
+            flow_pretrained_model = "flow_imagenet"
+
+            if att:
+                logging.info("Using {}".format(attention))
+                feature_network = se_i3d_joint(
+                    rgb_pt=rgb_pretrained_model,
+                    flow_pt=flow_pretrained_model,
+                    num_classes=num_classes,
+                    attention=attention,
+                    pretrained=True,
+                )
+            else:
+                logging.info("No SELayer.")
+                feature_network = i3d_joint(
+                    rgb_pt=rgb_pretrained_model, flow_pt=flow_pretrained_model, num_classes=num_classes, pretrained=True
+                )
+            class_feature_dim = 2048
+            dmn_feature_dim = class_feature_dim / 2
+
+        elif model_name == "R3D_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_r3d(rgb=True, flow=True, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = r3d(rgb=True, flow=True, pretrained=True)
+            class_feature_dim = 1024
+            dmn_feature_dim = class_feature_dim / 2
+
+        elif model_name == "R2PLUS1D_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                feature_network = se_r2plus1d(rgb=True, flow=True, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = r2plus1d(rgb=True, flow=True, pretrained=True)
+            class_feature_dim = 1024
+            dmn_feature_dim = class_feature_dim / 2
+
+        elif model_name == "MC3_18":
+            if att:
+                logging.info("{} using {}".format(model_name, attention))
+                # feature_network = se_mc3(rgb=False, flow=True, pretrained=True, attention=attention)
+            else:
+                logging.info("No SELayer.")
+                feature_network = mc3(rgb=True, flow=True, pretrained=True)
+            class_feature_dim = 1024
+            dmn_feature_dim = class_feature_dim / 2
+
+        else:
+            raise ValueError("Unsupported model: {}".format(model_name))
+
+    else:
+        raise ValueError("Input modality is not in [rgb, flow, joint]. Current is {}".format(image_modality))
+    return feature_network, int(class_feature_dim), int(dmn_feature_dim)
```

Comment:
1. The above is generic and should be in core API
2. Consider a more structured way to code it. The current controls are almost in the most inefficient ways, leading to lots of repetitions. Consider to use function argument and more specific modules to make the code more compact. Good examples are [digit access](https://github.com/pykale/pykale/blob/master/kale/loaddata/digits_access.py) and see https://krokotsch.eu/cleancode/2020/08/11/Unit-Tests-for-Deep-Learning.html how to write the repetitions in `test_augmentation_active_train_data` and `test_augmentation_active_test_data` into `_check_augmentation`